src/main/scala/com/lucidworks/spark/PartitionByTimeAssignmentStrategy.scala

```diff
@@ -0,0 +1,191 @@
+package com.lucidworks.spark
+
+import java.text.{ParseException, SimpleDateFormat}
+import java.util.{Collections, Date, TimeZone}
+
+import com.lucidworks.spark.util.{SolrQuerySupport, SolrRelationUtil, SolrSupport}
+import org.apache.solr.client.solrj.SolrQuery
+import com.lucidworks.spark.util.QueryConstants._
+
+import scala.util.control._
+import java.util.regex.Pattern
+
+import scala.collection.JavaConversions._
+import java.util.regex.Matcher
+import java.util.TimeZone
+
+import org.apache.lucene.queryparser.flexible.standard.StandardQueryParser
+import org.apache.lucene.search.{Query, TermRangeQuery}
+import org.apache.lucene.util.BytesRef
+import org.apache.solr.util.DateFormatUtil
+import org.apache.spark.Logging
+
+
+/**
+  * Created by akashmehta on 6/16/16.
+  */
+class PartitionByTimeAssignmentStrategy(val feature: PartitionByTimeFeature,val conf: SolrConf) extends Logging{
```

Comment:
I think "AssignmentStrategy" is not the right concept here. That's from Fusion and mainly applies to indexing. Please rename this and the corresponding test to: PartitionByTimeQuerySupport